app/controllers/passwords_controller.rb

```diff
@@ -24,7 +24,7 @@ def require_no_authentication
 
   def update
     super do |resource|
-      if resource.valid?
+      if resource.errors.empty?
```

Comment:
Why was this change required?

Reply:
(copied from https://github.com/alphagov/signon/pull/589/commits/be807797240debf08cee35d78bbb75b0c5640c36)

Due to a change when updating devise (https://github.com/plataformatec/devise/pull/4456/files)
blank passwords no longer get set on the resource when trying to reset
the password with a blank string. This change meant that the call to
resource.valid? now returned true rather than false, as the password was
nil - which is valid.

To fix this, the update method now checks for the presence of errors
rather than checking if the resource is valid.

Additionally, the error regarding password strength is no longer
displayed when submitting a blank password so that test has been removed

Reply:
Eeuw... that's a bit unfortunate that it's not valid but it is. Not much else you can do I don't suppose.